File: design_patterns/behavioural/template_method/go/internal/documents/word.go
package documents

import "fmt"

type WordDocument struct{}

func (d *WordDocument) Open() {
	fmt.Println("Opening WORD document")
}

func (d *WordDocument) Parse() {
	fmt.Println("Opening WORD document")
}

func (d *WordDocument) Export() {
	fmt.Println("Export WORD document")
}

func (d *WordDocument) Close() {
	fmt.Println("Close WORD document")
}

File: design_patterns/behavioural/template_method/python/src/template_method/documents/word.py
from .document import Document


class WORDDocument(Document):
    def Open(self):
        print("Opening WORD document")

    def Parse(self):
        print("Parsing WORD document")

    def Export(self):
        print("Exporting WORD document")

    def Close(self):
        print("Closing WORD document")
